src/audio/dai-zephyr.c

```diff
@@ -1669,7 +1669,7 @@ static int dai_copy(struct comp_dev *dev)
 int dai_common_ts_config_op(struct dai_data *dd, struct comp_dev *dev)
 {
 	struct ipc_config_dai *dai = &dd->ipc_config;
-	struct dai_ts_cfg cfg;
+	struct dai_ts_cfg *cfg = (struct dai_ts_cfg *)&dd->ts_config;
```

Comment:
why do we actually have `struct dai_ts_cfg` identical to `struct timestamp_cfg` and `struct dai_ts_data` identical to `struct timestamp_data`?

Reply:
The content of both struct is the same, and the final function in zephyr/driver/dai/intel/ needs such input parameter.

Reply:
right, so would it be possible to drop `struct timestamp_cfg` and `struct timestamp_data` completely and use `struct dai_ts_cfg` and `struct dai_ts_data` instead everywhere?

Reply:
okay, then I would like to redefine the struct struct dai_data.

Reply:
yes, I think that would be good - and define copies of those structures in dai-legacy.h under `#ifndef __ZEPHYR__`

Reply:
but if we are using dai_legacy.c, we will use the original struct timestamp_cfg and struct timestamp_data(because the original interface without zephyr needs such input parameters). So I don't think it is good to redefine struct dai_data in dai-legacy.h in this PR.

Reply:
sure, we can rename them for legacy, right? And no, it doesn't have to be fixed in this PR

Reply:
then I'd like use another PR to redefine for legacy, because I can not test it before I push code to PR right now.